conan/tools/env/environment.py

```diff
@@ -59,6 +59,7 @@ def environment_wrap_command(conanfile, env_filenames, env_folder, cmd, subsyste
         launchers = " && ".join('"{}"'.format(b) for b in bats)
         if ps1s:
             ps1_launchers = f'{powershell} -Command "' + " ; ".join('&\'{}\''.format(f) for f in ps1s)
```

Comment:
Hello @czoido 

I am pretty sure the `.join('&\'{}\''.format(f) for f in ps1s)` could be changed as well to `.join('{}'.format(f) for f in ps1s)`
The minimal local tests I have run seem to go well with this change and it does improve readability in my opinion
I think the `&` syntax is not necessary with the `-Command` anymore. Am I mistaken?

Reply:
Hi @Todiq,
Removing the & when invoking the scripts causes that the environment variables that should be injected from the ps1 files will not be applied correctly in the execution contexts, so I'm afraid those are necessary.
Is it possible to give us an small reproducible example to be able to add a test to check what this PR is fixing?

Reply:
Just added some tests.
Reproducible case is here : https://github.com/conan-io/conan-center-index/issues/26187

Reply:
Thank you so much, I could reproduce the issue, let me check what you added and simplify a bit.